modules/util/source-formatter/src/main/java/com/liferay/source/formatter/check/UpgradeGetClassNamesMethodCheck.java

```diff
@@ -0,0 +1,62 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.source.formatter.check;
+
+import com.liferay.portal.kernel.util.StringUtil;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * @author Tamyris Bernardo
+ */
+public class UpgradeGetClassNamesMethodCheck
+	extends BaseUpgradeMatcherReplacementCheck {
+
+	@Override
+	protected String afterFormat(
+		String fileName, String absolutePath, String content,
+		String newContent) {
+
+		if (fileName.endsWith(".java")) {
+			newContent = addNewImports(newContent);
+		}
+
+		return newContent;
+	}
+
+	@Override
+	protected String formatMatcherIteration(
+		String content, String newContent, Matcher matcher) {
+
+		String methodCall = matcher.group();
+
+		if (!hasClassOrVariableName("Indexer", content, content, methodCall)) {
+			return content;
+		}
+
+		return StringUtil.replace(
+			newContent, methodCall,
+			StringUtil.replace(
+				methodCall, "getClassNames", "getSearchClassNames"));
+	}
+
+	@Override
+	protected String[] getNewImports() {
+		return new String[] {"com.liferay.portal.kernel.search.Indexer"};
+	}
```

Comment:
Hi @TamyrisTorres, why do you add this import? If the trigger is the Indexer class the import should be already there, no?